File: src/main/java/com/github/squirrelgrip/meter/service/Line900Processor.java
package com.github.squirrelgrip.meter.service;

import com.github.squirrelgrip.meter.database.DatabaseSession;
import com.github.squirrelgrip.meter.domain.ContextKey;
import com.github.squirrelgrip.meter.exception.VersionNotSetException;

import java.util.Map;

import static com.github.squirrelgrip.meter.domain.ContextKey.*;

public class Line900Processor implements LineProcessor {
    @Override
    public void preProcess(Map<ContextKey, Object> context) {
        if (!context.containsKey(ContextKey.VERSION)) {
            throw new VersionNotSetException(context);
        }
        context.remove(NMI);
        context.remove(INTERVAL_LENGTH);
    }

    @Override
    public void process(String[] tokens, Map<ContextKey, Object> context) {
        context.remove(VERSION);

        DatabaseSession databaseSession = (DatabaseSession) context.get(SESSION);
        databaseSession.commit();
    }
}

File: src/main/kotlin/com/github/squirrelgrip/meter/service/Line900Processor.kt
package com.github.squirrelgrip.meter.service

import com.github.squirrelgrip.meter.domain.ContextKey
import com.github.squirrelgrip.meter.domain.MeterContext

class Line900Processor(): BaseLineProcessor() {

    override fun process(
        context: MeterContext,
        tokens: Array<String>
    ) {
        context.remove(ContextKey.NMI)
        context.remove(ContextKey.INTERVAL_LENGTH)
        context.remove(ContextKey.VERSION)
        context.session.commit()
    }
}
